src/ethernet_interface.cpp

```diff
@@ -858,7 +858,8 @@ void EthernetInterface::writeConfigurationFile()
         lla.emplace_back("no");
 #endif
         network["IPv6AcceptRA"].emplace_back(ipv6AcceptRA() ? "true" : "false");
-        if (dhcp6())
+        if ((std::string(ENABLE_DHCP6_WITHOUT_RA) == std::string("solicit")) &&
```

Comment:
if dhcp6() is not a costly check, bring that as the first check in the "&&" . you can avoid creating two-string objects. Just give a string constructor call to only one of them . I think overload resolution will work just with that.

Reply:
Updated code, now checking dhcp6() at first place.